src/components/Articles.jsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import * as SiteActions from '../actions/siteActions';
+import ArticleStore from '../stores/ArticleStore';
+import SingleArticle from './SingleArticle.jsx';
+import Sidebar from './Sidebar.jsx';
+
+/**
+ * @class Articles
+ * @extends {React.Component}
+ */
+class Articles extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      articles: [],
+      loading: true,
+      sortBy: [],
+    };
+    this.sortArticles = this.sortArticles.bind(this);
+    this.fetchArticles = this.fetchArticles.bind(this);
+    this.SortArticleBy = [];
+  }
+
+  /**
+   * @method componentDidMount - Runs after the page has been rendered
+   * @return {void}
+   * Retrieves article sort value by calling getSortOptions()
+   * and assigns it to instance variable
+   * Makes a call to get list of Articles
+   * Listens for a change in the ArticleStore
+   */
+  componentDidMount() {
+    this.SortArticleBy = this.getSortOptions()[0];
+    ArticleStore.on('change', this.fetchArticles);
+    SiteActions.fetchArticles(this.props.match.params.source, this.SortArticleBy);
+  }
+
+  /**
+   * @method componentWillReceiveProps - Runs after page has been rendered and props have changed
+   * @return {void}
+   * Retrieves article sort value by calling getSortOptions()
+   * and assigns it to instance variable
+   * Makes a call to get list of updated Articles
+   * Sets the state with new
+   */
+  componentWillReceiveProps(props) {
+    this.SortArticleBy = this.getSortOptions()[0];
+    SiteActions.fetchArticles(props.match.params.source, this.SortArticleBy);
+    this.setState({
+      loading: true,
+      sortBy: this.getSortOptions(this.props),
+      articles: ArticleStore.getArticles(),
+    });
+  }
+
+  /**
+   * @method componentWillUnmount - Runs before component is removed from the DOM
+   * @return {void}
+   * Removes the change Listener from ArticleStore
+   */
+  componentWillUnmount() {
```

Comment:
componentWillMount was never tested too. can this be added to what need to be tested..?